### DISCUSSION TITLE
Sergen and enums in different assemblies

### DISCUSSION THREAD

**Discussion by enriquin** [serenity-is/Serenity#6972] [Help] [answered]
Hello,
My row classes should contain fields of type "enum" defined in a shared assembly between multiple solutions.

The problem is that sergen does not generate the typescript files for the enums in the "external assembly" it puts the references to the typescript files in my generated row typescript files but the files are not created and the types are not generated.

Is there any way to instruct sergen to search in additional assemblies? Any idea on how to do that?

Thanks in advance!!
Enrique

**Comment by volkanceylan** [comment, marked as answer]
Normally, TS typings for a separate assembly should be generated at the source project, not the referencing project. See common-features samples, for example:

https://github.com/serenity-is/common-features/blob/master/src/Serenity.Demo.Northwind/Modules/Order/OrderShippingState.cs

The typing for OrderShippingState is generated in Serenity.Demo.Northwind, and referenced from BasicSamples etc.

But if you want to go easy and include that assembly in code generation for your project, without going "feature project" route, you may do that in sergen.json (not so recommended):

https://github.com/serenity-is/Serenity/blob/master/src/Serenity.Net.CodeGenerator/Generator/GeneratorConfig.cs

```json
{
    "ServerTypings": {
      "Assemblies": [
        "bin/Debug/net6.0/MyProject.Web.dll",
        "bin/Debug/net6.0/MyEnums..dll",
      ]
    }
}
```

Fix the paths based on your project.
